opsgenie/resource_opsgenie_heartbeat.go

```diff
@@ -112,8 +113,11 @@ func resourceOpsgenieHeartbeatRead(d *schema.ResourceData, meta interface{}) err
 	}
 
 	result, err := client.Get(context.Background(), d.Id())
-	if err != nil {
-		return err
+	if isNotFoundError(err) {
+		d.SetId("")
```

Comment:
This seems to be the way to mark a resource as missing.

> You can also signal existence in the Read method by calling d.SetId("") if the Resource is no longer present and should be removed from state.

source: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema?tab=doc#Resource